be/src/exprs/vectorized/array_map_expr.cpp

```diff
@@ -0,0 +1,122 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+#include "exprs/vectorized/array_map_expr.h"
+
+#include "column/array_column.h"
+#include "column/chunk.h"
+#include "column/column_helper.h"
+#include "column/const_column.h"
+#include "column/fixed_length_column.h"
+#include "column/vectorized_fwd.h"
+#include "exprs/anyval_util.h"
+#include "exprs/expr_context.h"
+#include "exprs/vectorized/builtin_functions.h"
+#include "exprs/vectorized/function_helper.h"
+#include "exprs/vectorized/lambda_function.h"
+#include "gutil/strings/substitute.h"
+#include "runtime/current_thread.h"
+#include "runtime/user_function_cache.h"
+
+namespace starrocks::vectorized {
+ArrayMapExpr::ArrayMapExpr(const TExprNode& node) : Expr(node, false) {}
+
+inline bool offsets_equal(UInt32Column::Ptr array1, UInt32Column::Ptr array2) {
+    if (array1->size() != array2->size()) {
+        return false;
+    }
+    auto data1 = array1->get_data();
+    auto data2 = array2->get_data();
+    return std::equal(data1.begin(), data1.end(), data2.begin());
+}
+
+// The input array column maybe nullable, so first remove the wrap of nullable property.
+// The result of lambda expressions does not change the offsets of the current array and the null map.
+ColumnPtr ArrayMapExpr::evaluate(ExprContext* context, Chunk* ptr) {
+    std::vector<ColumnPtr> inputs;
+    NullColumnPtr input_null_map = nullptr;
+    std::shared_ptr<ArrayColumn> input_array = nullptr;
+    // for many valid arguments:
+    // if one of them is a null literal, the result is a null literal;
+    // if one of them is only null, then results are null;
+    // unfold const columns.
+    // make sure all inputs have the same offsets.
+    // TODO(fzh): support several arrays with different offsets and set null for non-equal size of arrays.
+    for (int i = 1; i < _children.size(); ++i) {
+        ColumnPtr child_col = EVALUATE_NULL_IF_ERROR(context, _children[i], ptr);
```

Comment:
suggest use chunk or chk instead of ptr.

Reply:
ok